fields/types/html/HtmlField.js

```diff
@@ -15,53 +16,40 @@ function getId () {
 	return 'keystone-html-' + lastId++;
 }
 
+// class HtmlField extends React.Component {
 module.exports = Field.create({
-
 	displayName: 'HtmlField',
 
 	getInitialState () {
-		return {
-			id: getId(),
-			isFocused: false,
-		};
-	},
-
-	initWysiwyg () {
-		if (!this.props.wysiwyg) return;
-
-		var self = this;
-		var opts = this.getOptions();
-
-		opts.setup = function (editor) {
-			self.editor = editor;
-			editor.on('change', self.valueChanged);
-			editor.on('focus', self.focusChanged.bind(self, true));
-			editor.on('blur', self.focusChanged.bind(self, false));
-		};
-
-		this._currentValue = this.props.value;
-		tinymce.init(opts);
-	},
-
-	componentDidUpdate (prevProps, prevState) {
-		if (prevState.isCollapsed && !this.state.isCollapsed) {
-			this.initWysiwyg();
+		// convert saved editor content into the editor state
+		let editorState;
+		try {
+			// const formData = JSON.parse(this.props.value);
+			const formData = this.props.value;
+			if (formData.draft && formData.html && formData.html !== '') {
+				// create an EditorState from the raw Draft data
+				let contentState = Draft.ContentState.createFromBlockArray(Draft.convertFromRaw(formData.draft));
+				editorState = Draft.EditorState.createWithContent(contentState);
+			} else {
+				// create empty draft object
+				editorState = EditorState.createEmpty();
+			}
 		}
-
-		if (_.isEqual(this.props.dependsOn, this.props.currentDependencies)
-			&& !_.isEqual(this.props.currentDependencies, prevProps.currentDependencies)) {
-			var instance = tinymce.get(prevState.id);
-			if (instance) {
-				tinymce.EditorManager.execCommand('mceRemoveEditor', true, prevState.id);
-				this.initWysiwyg();
+		catch (error) {
+			// transform existing HTML content produced by TinyMCE
+			if (this.props.value && typeof (this.props.value) === 'string') {
+				// create an EditorState from HTML
+				editorState = EditorState.createWithContent(ContentState.createFromBlockArray(processHTML(this.props.value)));
 			} else {
-				this.initWysiwyg();
+				// create empty EditorState
+				editorState = EditorState.createEmpty();
```

Comment:
If we use Draft from the scratch, maybe we don't need to care about those posts made by tinyMCE.